src/ApiConnector.php

```diff
@@ -21,15 +30,21 @@ class ApiConnector
 	 * @param string $secret     Your Penneo API secret
 	 * @param string $endpoint   The API endpoint url. This defaults to the API sandbox.
 	 */
-	public static function initialize($key, $secret, $endpoint=null, $user=null, $headers=null)
+	public static function initialize($key, $secret, $endpoint=null, $user=null, array $headers=null)
 	{
-		if ($endpoint)
-			self::$endpoint = $endpoint;
-		if ($headers)
+		self::$endpoint = $endpoint ?: self::getDefaultEndpoint();
+
+		self::$headers = self::getDefaultHeaders();
+		if ($headers) {
 			self::$headers = array_merge($headers, self::$headers);
-		if ($user)
+		}
+
+		if ($user) {
 			self::$headers['penneo-api-user'] = intval($user);
-		
+		} else {
+			unset(self::$headers['penneo-api-user']);

```

Comment:
This else section is not really needed since you overwrote any previous headers in line 37.